pkg/es/config/config.go

```diff
@@ -299,6 +305,14 @@ func (c *Configuration) GetIndexRolloverFrequencyServicesDuration() time.Duratio
 	return -24 * time.Hour
 }
 
+// GetIndexRolloverFrequencySamplingDuration returns jaeger-sampling index rollover frequency duration
+func (c *Configuration) GetIndexRolloverFrequencySamplingDuration() time.Duration {
+	if c.IndexRolloverFrequencySampling == "hour" {
+		return -1 * time.Hour
+	}
+	return -24 * time.Hour
+}
```

Comment:
you can refactor all 3 functions to invoke a shared helper that would convert a frequency string to Duration.